File: 001_Two_Sum/solution.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dt = {}
        for i in range(len(nums)):
            if target - nums[i] in dt:
                return [dt[target-nums[i]], i]
            dt[nums[i]] = i
        return []

# Runtime: 62ms

File: 001_Two_Sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var m = new Map();
    for ( i = 0; i < nums.length; i++) {
        if (m.has(target - nums[i])) {
            return [m.get(target - nums[i]), i];
        }
        m.set(nums[i], i);
    }
    return [];
};



// Runtime: 152 ms